mithril-common/src/entities/signer.rs

```diff
@@ -1,25 +1,60 @@
-use crate::entities::{PartyId, Stake};
+use crate::{
+    crypto_helper::KESPeriod,
+    entities::{HexEncodedKey, PartyId, Stake},
+};
+
 use serde::{Deserialize, Serialize};
 use sha2::{Digest, Sha256};
 
 /// Signer represents a signing participant in the network
 #[derive(Clone, Debug, PartialEq, Eq, Default, Serialize, Deserialize)]
 pub struct Signer {
     /// The unique identifier of the signer
+    // TODO: Should be removed once the signer certification is fully deployed
     #[serde(rename = "party_id")]
     pub party_id: PartyId,
 
     /// The public key used to authenticate signer signature
     #[serde(rename = "verification_key")]
-    pub verification_key: String,
+    pub verification_key: HexEncodedKey,
+
+    /// The encoded signer 'Mithril verification key' signature (signed by the Cardano node KES secret key)
+    // TODO: Option should be removed once the signer certification is fully deployed
+    #[serde(
+        rename = "verification_key_signature",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub verification_key_signature: Option<HexEncodedKey>,
+
+    /// The encoded operational certificate of stake pool operator attached to the signer node
+    // TODO: Option should be removed once the signer certification is fully deployed
+    #[serde(
+        rename = "operational_certificate",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub operational_certificate: Option<HexEncodedKey>,
+
+    /// The kes period used to compute the verification key signature
+    // TODO: This kes period shoud not be used as is and should probably be within an allowed range of kes period for the epoch
+    #[serde(rename = "kes_period", skip_serializing_if = "Option::is_none")]
+    pub kes_period: Option<KESPeriod>,
 }
 
 impl Signer {
     /// Signer factory
-    pub fn new(party_id: PartyId, verification_key: String) -> Signer {
+    pub fn new(
+        party_id: PartyId,
+        verification_key: String,
+        verification_key_signature: Option<HexEncodedKey>,
+        operational_certificate: Option<HexEncodedKey>,
```

Comment:
Is it ok that we use `HexEncoddedKey` type for the vk signature and the opcert? Seems confusing (but I guess its just another way of saying `String`)

Reply:
I have replaced each hex encoded type previously represented with `HexEncodedKey` with a dedicated type that is an alias of `HexEncodedKey`, such as `HexEncodedOpCert` :+1: